build_tools/packaging/linux/build_package.py

```diff
@@ -58,6 +64,56 @@ class PackageConfig:
 
 ################### Debian package creation #######################
 def create_deb_package(pkg_name, config: PackageConfig):
+    """Function to create deb package
+    Invoke the creation of versioned and non versioned package
+    Move the DEB packages to destination directory
+
+    Parameters:
+    pkg_name : Name of the package to be created
+    config: Configuration object containing package metadata
+
+    Returns: None
+    """
+    print(inspect.currentframe().f_code.co_name)
+    print(f"Package Name: {pkg_name}")
+
+    create_nonversioned_deb_package(pkg_name, config)
+    create_versioned_deb_package(pkg_name, config)
+    move_packages_to_destination(pkg_name, config)
+
+
+def create_nonversioned_deb_package(pkg_name, config: PackageConfig):
+    """Function to create non versioned deb package
+    Get package details and generate control file
+    Non versioned Packages are meta packages.
+    Create deb package
+
+    Parameters:
+    pkg_name : Name of the package to be created
+    config: Configuration object containing package metadata
+
+    Returns: None
+    """
+    print(inspect.currentframe().f_code.co_name)
```

Comment:
~~This print is missing context. Is it leftover from some debugging? Do want to keep it, and if so can you add more text preceding it? I see a few similar prints further down in the file.~~

Edit: ooooh wait, this is the "current function name" or something?

* https://docs.python.org/3/library/inspect.html#inspect.currentframe
* https://stackoverflow.com/questions/5067604/determine-function-name-from-within-that-function

That was not obvious to me. Is it worth keeping that generic, or could these prints be replaced with explicit code like
```python
  print("Creating nonversioned deb package")
```

If you want to keep the generic code, how about using a helper function/lambda like so https://stackoverflow.com/a/31615605 ?

Reply:
print(inspect.currentframe().f_code.co_name) . This will print the  current function name
Let me check the stackoverlfow link.  It  states get rame is faster than the inspect method

Reply:
Implemented a common function as mentioned in the stackoverflow link